# HG changeset patch
# User Torsten <ttl@justmail.de>
# Date 1369572581 -7200
#      Sun May 26 14:49:41 2013 +0200
# Node ID 553cfdd5d66082209187c4f46d2ef040f7720ee4
# Parent  ccfbc767ff5a38519b3591d5ea805d083d65175c
make white space visibility in the editor configurable

* settings-dialog.ui: insert checkboxes for white space visibility
* settings-dialog.cc(read_settings): read state for checkboxes from settings,
  (write_changed_settings): write state of checkboxes into settings file
* file-editor-tab.cc(notice-settings): load white space visibility from settings

diff --git a/libgui/src/m-editor/file-editor-tab.cc b/libgui/src/m-editor/file-editor-tab.cc
--- a/libgui/src/m-editor/file-editor-tab.cc
+++ b/libgui/src/m-editor/file-editor-tab.cc
@@ -1118,16 +1118,24 @@ file_editor_tab::notice_settings (const 
       _edit_area->setAutoCompletionCaseSensitivity
         (settings->value ("editor/codeCompletion_case",true).toBool ());
       _edit_area->setAutoCompletionThreshold
         (settings->value ("editor/codeCompletion_threshold",2).toInt ());
     }
   else
     _edit_area->setAutoCompletionThreshold (-1);
 
+  if (settings->value ("editor/show_white_space",false).toBool ())
+    if (settings->value ("editor/show_white_space_indent",false).toBool ())
+      _edit_area->setWhitespaceVisibility (QsciScintilla::WsVisibleAfterIndent);
+    else
+      _edit_area->setWhitespaceVisibility (QsciScintilla::WsVisible);
+  else
+    _edit_area->setWhitespaceVisibility (QsciScintilla::WsInvisible);
+
   if (settings->value ("editor/showLineNumbers", true).toBool ())
     {
       _edit_area->setMarginLineNumbers (2, true);
       _edit_area->setMarginWidth (2, lexer_font_metrics.width ("9999"));
     }
   else
     {
       _edit_area->setMarginLineNumbers (2, false);
diff --git a/libgui/src/settings-dialog.cc b/libgui/src/settings-dialog.cc
--- a/libgui/src/settings-dialog.cc
+++ b/libgui/src/settings-dialog.cc
@@ -103,16 +103,18 @@ settings_dialog::settings_dialog (QWidge
   ui->editor_highlightCurrentLine->setChecked (settings->value ("editor/highlightCurrentLine",true).toBool () );
 
   ui->editor_codeCompletion->setChecked (settings->value ("editor/codeCompletion",true).toBool () );
   ui->editor_spinbox_ac_threshold->setValue (settings->value ("editor/codeCompletion_threshold",2).toInt ());
   ui->editor_checkbox_ac_keywords->setChecked (settings->value ("editor/codeCompletion_keywords",true).toBool ());
   ui->editor_checkbox_ac_document->setChecked (settings->value ("editor/codeCompletion_document",false).toBool ());
   ui->editor_checkbox_ac_case->setChecked (settings->value ("editor/codeCompletion_case",true).toBool ());
   ui->editor_checkbox_ac_replace->setChecked (settings->value ("editor/codeCompletion_replace",false).toBool ());
+  ui->editor_ws_checkbox->setChecked (settings->value ("editor/show_white_space",false).toBool ());
+  ui->editor_ws_indent_checkbox->setChecked (settings->value ("editor/show_white_space_indent",false).toBool ());
   ui->editor_longWindowTitle->setChecked (settings->value ("editor/longWindowTitle",false).toBool ());
   ui->editor_restoreSession->setChecked (settings->value ("editor/restoreSession",true).toBool ());
   ui->terminal_fontName->setCurrentFont (QFont (settings->value ("terminal/fontName","Courier New").toString()) );
   ui->terminal_fontSize->setValue (settings->value ("terminal/fontSize",10).toInt ());
   ui->showFileSize->setChecked (settings->value ("filesdockwidget/showFileSize",false).toBool());
   ui->showFileType->setChecked (settings->value ("filesdockwidget/showFileType",false).toBool());
   ui->showLastModified->setChecked (settings->value ("filesdockwidget/showLastModified",false).toBool());
   ui->showHiddenFiles->setChecked (settings->value ("filesdockwidget/showHiddenFiles",false).toBool());
@@ -406,16 +408,18 @@ settings_dialog::write_changed_settings 
   settings->setValue ("editor/highlightCurrentLine", ui->editor_highlightCurrentLine->isChecked ());
   settings->setValue ("editor/highlight_current_line_color",_editor_current_line_color->color ());
   settings->setValue ("editor/codeCompletion", ui->editor_codeCompletion->isChecked ());
   settings->setValue ("editor/codeCompletion_threshold", ui->editor_spinbox_ac_threshold->value ());
   settings->setValue ("editor/codeCompletion_keywords", ui->editor_checkbox_ac_keywords->isChecked ());
   settings->setValue ("editor/codeCompletion_document", ui->editor_checkbox_ac_document->isChecked ());
   settings->setValue ("editor/codeCompletion_case", ui->editor_checkbox_ac_case->isChecked ());
   settings->setValue ("editor/codeCompletion_replace", ui->editor_checkbox_ac_replace->isChecked ());
+  settings->setValue ("editor/show_white_space", ui->editor_ws_checkbox->isChecked ());
+  settings->setValue ("editor/show_white_space_indent", ui->editor_ws_indent_checkbox->isChecked ());
   settings->setValue ("editor/longWindowTitle", ui->editor_longWindowTitle->isChecked());
   settings->setValue ("editor/restoreSession", ui->editor_restoreSession->isChecked ());
   settings->setValue ("terminal/fontSize", ui->terminal_fontSize->value());
   settings->setValue ("terminal/fontName", ui->terminal_fontName->currentFont().family());
   settings->setValue ("filesdockwidget/showFileSize", ui->showFileSize->isChecked ());
   settings->setValue ("filesdockwidget/showFileType", ui->showFileType->isChecked ());
   settings->setValue ("filesdockwidget/showLastModified", ui->showLastModified->isChecked ());
   settings->setValue ("filesdockwidget/showHiddenFiles", ui->showHiddenFiles->isChecked ());
diff --git a/libgui/src/settings-dialog.ui b/libgui/src/settings-dialog.ui
--- a/libgui/src/settings-dialog.ui
+++ b/libgui/src/settings-dialog.ui
@@ -203,16 +203,50 @@
             <bool>true</bool>
            </property>
            <property name="text">
             <string>Show line numbers</string>
            </property>
           </widget>
          </item>
          <item>
+          <layout class="QHBoxLayout" name="horizontalLayout_4">
+           <item>
+            <widget class="QCheckBox" name="editor_ws_checkbox">
+             <property name="text">
+              <string>Show white space</string>
+             </property>
+            </widget>
+           </item>
+           <item>
+            <widget class="QCheckBox" name="editor_ws_indent_checkbox">
+             <property name="enabled">
+              <bool>false</bool>
+             </property>
+             <property name="text">
+              <string>But no white spaces used for indentation</string>
+             </property>
+            </widget>
+           </item>
+           <item>
+            <spacer name="horizontalSpacer_12">
+             <property name="orientation">
+              <enum>Qt::Horizontal</enum>
+             </property>
+             <property name="sizeHint" stdset="0">
+              <size>
+               <width>40</width>
+               <height>20</height>
+              </size>
+             </property>
+            </spacer>
+           </item>
+          </layout>
+         </item>
+         <item>
           <layout class="QGridLayout" name="editor_grid_current_line" rowstretch="0" columnstretch="0,0,0,0,0">
            <item row="0" column="0">
             <widget class="QCheckBox" name="editor_highlightCurrentLine">
              <property name="enabled">
               <bool>true</bool>
              </property>
              <property name="text">
               <string>Highlight current line</string>
@@ -452,28 +486,28 @@
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
-        <widget class="QCheckBox" name="useCustomFileEditor">
-         <property name="enabled">
-          <bool>true</bool>
-         </property>
-         <property name="text">
-          <string>Use custom file editor</string>
-         </property>
-        </widget>
-       </item>
-       <item>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
+          <widget class="QCheckBox" name="useCustomFileEditor">
+           <property name="enabled">
+            <bool>true</bool>
+           </property>
+           <property name="text">
+            <string>Use custom file editor</string>
+           </property>
+          </widget>
+         </item>
+         <item>
           <widget class="QLabel" name="customEditorLabel">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Command  line (%f=file, %l=line):</string>
            </property>
           </widget>
@@ -1259,10 +1293,26 @@
      <y>172</y>
     </hint>
     <hint type="destinationlabel">
      <x>525</x>
      <y>203</y>
     </hint>
    </hints>
   </connection>
+  <connection>
+   <sender>editor_ws_checkbox</sender>
+   <signal>toggled(bool)</signal>
+   <receiver>editor_ws_indent_checkbox</receiver>
+   <slot>setEnabled(bool)</slot>
+   <hints>
+    <hint type="sourcelabel">
+     <x>85</x>
+     <y>119</y>
+    </hint>
+    <hint type="destinationlabel">
+     <x>282</x>
+     <y>119</y>
+    </hint>
+   </hints>
+  </connection>
  </connections>
 </ui>
